juriscraper/pacer/email.py

```diff
@@ -534,116 +534,55 @@ def _get_docket_entries(
-            # Remove docket number traces "-AAA"
-            # Remove CH after docket and BK after short description for dcb
-            regex = r"^-.*?\s|C[Hh][\s\d]+|[ (]?B[Kk]( Other)?[) ]?"
-            short_description = re.sub(regex, "", short_description)
-            separator = ";" if self.court_id == "arwb" else "-"
-            short_description = short_description.rsplit(separator, 1)[0]
-        elif self.court_id == "nysb":
-            # In: 22-22507-cgm Ch13 Affidavit Re: Gerasimos Stefanitsis
-            # Out: Affidavit
-            short_description = subject.split(case_name)[0]
-            short_description = short_description.replace("Re:", "")
-            short_description = short_description.split(docket_number)[-1]
-
-            # Remove strings starting with "Ch" followed by a number
-            regex = r"\bCh\d+\b"
-            short_description = re.sub(regex, "", short_description)
-
-            # Remove docket number traces "-AAA"
-            regex = r"^-.*?\s"
-            short_description = re.sub(regex, "", short_description)
-        elif self.court_id in ["pawb", "ndb", "deb", "pamb", "nhb"]:
-            # In: Ch-7 22-20823-GLT U LOCK INC Reply
-            # Out: Reply
-            if case_name in subject:
-                short_description = subject.split(case_name)[-1]
-            elif case_name[:18] in subject:
-                # See deb_2.txt, pamb_1 and pamb_3 for examples
-                short_description = subject.split(case_name[:18])[-1]
-            elif (
-                " and " in case_name and case_name.split(" and ")[0] in subject
-            ):
-                # See pamb_2.txt
-                short_description = subject.split(case_name.split(" and ")[0])[
-                    -1
-                ]
-        elif self.court_id in [
-            "tnmb",
-        ]:
-            # In: Docket Order - Continue Hearing (Auto) Ch 13 Jeffery Wayne Lovell and Tiffany Nicole Lovell 1:24-bk-01377
-            # Out: Docket Order - Continue Hearing (Auto) Ch 13
-            if case_name in subject:
-                short_description = subject.split(case_name)[0]
+        for part in self.docket_numbers + self.case_names:
+            subject = subject.replace(part, " ")
+
+        # Sometimes the full case name is not used in the `subject`
+        # Some courts use a 18 character limit
+        # See deb_2.txt, pamb_1 and pamb_3 for examples
+        for case_name in self.case_names:
+            subject = subject.replace(case_name[:18].strip(), " ")
+            subject = subject.replace(case_name.split(" and ")[0], " ")
+
+        # Deletes:
+        # - extra docket number 'components', such as `federal_dn_judge_initials_assigned`
+        #     these are usually 3 letters long. However, we want to keep some special acronyms
+        #     such as MOR (Merchant of Record?)
+        # - "NEF: " placeholder
+        component_regex = r"((?!-MOR)(\-[A-Z]{2,}))|(\-[a-z]{2,})|(NEF:? )"
```

Comment:
I found a couple of examples where removing suffixes outside the exceptions listed here can be problematic:

For instance, this case where the subject contains:
`Order -Non-motion related-`

But it’s being parsed as:
`Order -Non related-`

[pawd_1.txt](https://github.com/user-attachments/files/18234736/pawd_1.txt)

![Screenshot 2024-12-23 at 7 26 51 p m](https://github.com/user-attachments/assets/08bbe8ed-9e5c-4839-a152-217bbad76b4b)

Or in the same case:
`Motion to Withdraw/Dismiss Document -bk-`

I couldn’t find an email example for this one, but I modified a subject and tested its parsing. The result was:
`Motion to Withdraw/Dismiss Document`

![Screenshot 2024-12-23 at 7 33 25 p m](https://github.com/user-attachments/assets/b5ea13dd-f062-49d8-a9b0-e23fa7d6ae00)

So this also seems difficult in determining when these suffixes should be removed or which ones should remain as exceptions. Perhaps we should just consider adding exceptions as we encounter them?

This seems easier maybe keeping the suffixes in parentheses works?

[cacb.txt](https://github.com/user-attachments/files/18234809/cacb.txt)
![Screenshot 2024-12-23 at 7 48 06 p m](https://github.com/user-attachments/assets/b519470f-2250-4c4e-b24e-0ff612609d9d)

`ORDER to continue/reschedule hearing (BNC-PDF)`

Currently the description is being parsed as:
`ORDER to continue/reschedule hearing (BNC )`